keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,1044 @@
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+The rollout will not impact already running workloads, unless they set the
+`managed-by` label to a custom value, but this would require a prior intentional
+action.
+
+###### What specific metrics should inform a rollback?
+
+A substantial increase in the `apiserver_request_total[code=409, resource=job, group=batch]`,
+while there are jobs with the custom `managed-by` label, can be indicative of
+the built-in job controller stepping onto another controller, causing conflicts.
+This can be further investigate per-job by checking the `.metadata.managedFields.manager`
+being flipped between two owners.
+
+The feature is opt-in so in case of such problems the custom `managed-by` label
+should not be used.
+
+A substantial drop in the `job_sync_duration_seconds`, while the number of
+jobs with the custom `managed-by` label is low, could be indicative of the
+Job controller skipping reconciliation of jobs it should reconcile. This could
+be further investigated per-job by looking at the timestamp of changes in
+`.metadata.managedFields.time`, and owners in `.metadata.managedFields.manager`.
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+It will be tested manually prior to beta launch.
+
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+Check if a Job has the `managed-by` label.
```

Comment:
This question is from the cluster-admin level, iow. don't expect admins to go through all the jobs looking for the label. They should have a higher-level information available. See my comment about exposing information when using external controller.

Reply:
Sure, added the answer using the newly proposed metric